plugins/parsers/dropwizard/parser.go

```diff
@@ -0,0 +1,239 @@
+	Separator string
+	Templates []string
+
+	templateEngine *templating.Engine
+}
+
+// Parse parses the input bytes to an array of metrics
+func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
+
+	metrics := make([]telegraf.Metric, 0)
+
+	metricTime := p.parseTime(buf)
+	dwr, err := p.unmarshalMetrics(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	metrics = p.readDWMetrics("counter", dwr["counters"], metrics, metricTime)
+	metrics = p.readDWMetrics("meter", dwr["meters"], metrics, metricTime)
+	metrics = p.readDWMetrics("gauge", dwr["gauges"], metrics, metricTime)
+	metrics = p.readDWMetrics("histogram", dwr["histograms"], metrics, metricTime)
+	metrics = p.readDWMetrics("timer", dwr["timers"], metrics, metricTime)
+
+	jsonTags := p.readTags(buf)
+
+	// fill json tags first
+	if len(jsonTags) > 0 {
+		for _, m := range metrics {
+			for k, v := range jsonTags {
+				// only set the tag if it doesn't already exist:
+				if !m.HasTag(k) {
+					m.AddTag(k, v)
+				}
+			}
+		}
+	}
+	// fill default tags last
+	if len(p.DefaultTags) > 0 {
+		for _, m := range metrics {
+			for k, v := range p.DefaultTags {
+				// only set the default tag if it doesn't already exist:
+				if !m.HasTag(k) {
+					m.AddTag(k, v)
+				}
+			}
+		}
+	}
+
+	return metrics, nil
+}
+
+// InitTemplating initializes the templating support
+func (p *Parser) InitTemplating() error {
+	if len(p.Templates) > 0 {
+		defaultTemplate, _ := templating.NewDefaultTemplateWithPattern("measurement*")
+		templateEngine, err := templating.NewEngine(p.Separator, defaultTemplate, p.Templates)
+		p.templateEngine = templateEngine
+		return err
+	}
+	return nil
+}
+
+// ParseLine is not supported by the dropwizard format
+func (p *Parser) ParseLine(line string) (telegraf.Metric, error) {
+	return nil, fmt.Errorf("ParseLine not supported: %s, for data format: dropwizard", line)
+}
+
+// SetDefaultTags sets the default tags
+func (p *Parser) SetDefaultTags(tags map[string]string) {
+	p.DefaultTags = tags
+}
+
+func (p *Parser) readTags(buf []byte) map[string]string {
+
+	if p.TagsPath != "" {
+		var tagsBytes []byte
+		tagsResult := gjson.GetBytes(buf, p.TagsPath)
+		if tagsResult.Index > 0 {
+			tagsBytes = buf[tagsResult.Index : tagsResult.Index+len(tagsResult.Raw)]
+		} else {
+			tagsBytes = []byte(tagsResult.Raw)
+		}
+		var tags map[string]string
+		err := json.Unmarshal(tagsBytes, &tags)
+		if err != nil {
+			log.Printf("W! failed to parse tags from json path '%s': %s\n", p.TagsPath, err)
+		} else if len(tags) > 0 {
+			return tags
+		}
+	}
+
+	tags := make(map[string]string)
+	for tagKey, jsonPath := range p.TagPathsMap {
+		tags[tagKey] = gjson.GetBytes(buf, jsonPath).String()
+	}
+	return tags
+}
+
+func (p *Parser) parseTime(buf []byte) time.Time {
+
+	if p.TimePath != "" {
+		timeFormat := p.TimeFormat
+		if timeFormat == "" {
+			timeFormat = time.RFC3339
+		}
+		time, err := time.Parse(timeFormat, gjson.GetBytes(buf, p.TimePath).String())
```

Comment:
I think it should be an error if the time path is not found, or if the time is wrong format.

Reply:
done